### ISSUE TITLE
Testing / Unit Testing

### ISSUE THREAD

**Issue by harperjohn** [BlazingSaddlesSolutions/TimeKeeper#10]
I like this idea... 

https://jonhilton.net/better-csharp-tests-lightbdd/?ck_subscriber_id=1036111148&utm_source=convertkit&utm_medium=email&utm_campaign=Tests+are+hard...%20-%203455564

**Comment by jonhilt** [comment]
Something else I've been wanting to look into (but haven't found time yet) is bUnit.

I've been hearing good things!

https://bunit.dev/index.html

**Comment by harperjohn** [comment]
Yes,  I have heard good things about bUnit too! ( and I was a bit surprised to see that LightBDD does not seem to support it yet... maybe it is just that there "readme" is out of date...)

**Comment by jonhilt** [comment]
Just seen this too (may or may not have any relevance for this project but I can think of times it might have been useful)

https://adamstorr.azurewebsites.net/blog/beginning-to-make-testing-with-httpclient-easier

**Comment by DotNetDublin** [comment]
Thanks for adding this issue as it's definitely something we need to implement.

Would LightBDD and bUnit not handle two different testing scenarios?

I can see using LightBDD for a traditional unit case scenario where I have a CalculateDeliveryCost function which accepts a post code as a parameter and I want to test that the expected delivery cost is being returned for a selection of post codes.

In bUnit I would be more testing that if the user clicks on the search button with "Apple" in the search text box then verify that the div with an ID of SearchResults is displaying the search results table consisting of specified html.

In some ways with bUnit we're testing both the back end and front end. Back end in terms of the search function returning expected results and the front end in terms of the results being displayed in a specified format.
